src/client/Microsoft.Identity.Client/Extensibility/ClientCredentialExtensionParameters.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Identity.Client.Extensibility
+{
+    /// <summary>
+    /// Provides application configuration context to client credential extensibility callbacks.
+    /// Contains read-only information about the confidential client application.
+    /// </summary>
+#if !SUPPORTS_CONFIDENTIAL_CLIENT
+    [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]  // hide confidential client on mobile
+#endif
+    public class ClientCredentialExtensionParameters
```

Comment:
Can we reuse AssertionRequestOptions ? Some of the properties don't match exactly, but it's ok if they are not set.